src/gzip/static_compression_handler.cr

```diff
@@ -0,0 +1,79 @@
+class Lucky::StaticCompressionHandler
+  Habitat.create do
+    setting serve_precompressed_assets : Bool = false
+  end
+
+  include HTTP::Handler
+
+  GZIP_FILE_EXTENSIONS = %w(.htm .html .txt .css .js .svg .json .xml .otf .ttf .woff .woff2)
```

Comment:
Nice! I was wondering about this. Could you make this a Habitat setting as well?

Reply:
I have to think about that. I'm not sure the size of the footgun. Mostly I'm thinking about binary that shouldn't be compressed. I'm wondering if it should be the framework's responsibility to know what is reasonable to compress? I'm a bit on the fence.

Reply:
@bdtomlin Oh sorry I think maybe I wasn't clear. I meant something like this:

```crystal
setting gzip_file_extensions = %w(.htm .html .txt .css .js .svg .json .xml .otf .ttf .woff .woff2)
```

So we still give a nice default, but users can change it or add to it if they want:

```crystal
StaticCompressionHandler.configure do |config|
  config.gzip_file_extensions << ".somethingelse"
end
```

Reply:
Yeah, I'm persuaded. Will do.